src/providers/ogr/qgsgeopackagedataitems.cpp

```diff
@@ -274,22 +280,22 @@ bool QgsGeoPackageConnectionItem::handleDrop( const QMimeData *data, Qt::DropAct
               }
             } );
 
-            QgsApplication::taskManager()->addTask( exportTask.release() );
           }
           else  // Import raster
           {
-
-            std::unique_ptr< QgsGeoPackageRasterWriterTask > exportTask( new QgsGeoPackageRasterWriterTask( dropUri, mPath ) );
+            QgsGeoPackageRasterWriterTask  *exportTask = new QgsGeoPackageRasterWriterTask( dropUri, mPath ) ;
+            mainTask->addSubTask( exportTask, importTasks );
+            importTasks << exportTask;
             // when export is successful:
-            connect( exportTask.get(), &QgsGeoPackageRasterWriterTask::writeComplete, this, [ = ]()
+            connect( exportTask, &QgsGeoPackageRasterWriterTask::writeComplete, this, [ = ]()
             {
               // this is gross - TODO - find a way to get access to messageBar from data items
```

Comment:
re:

> find a way to get access to the messageBar

Is `QgsVectorDataProvider::pushMessage` available here?

Unrelated, but still ;)

Reply:
Actually, I was thinking to rework `QgisApp::authMessageOut()` to make it more flexible and connect the slot to the (root) data items as I did with the `connectionsChanged` signals in the browser. 
What do you think?

Reply:
I was thinking about a centralized logging system inside QGIS which is available via `QgsApplication::logging`.
It acts as a central hub to which (text only) messages can be pushed and to which pieces like the message log, message bar or a file logger connect (hello server!).

```
QgsLogMessage {
  QString tag;
  QgsLogMessage::MessageLevel level;
  QString title;
  QString message;
  QString details;
  // Additional flags
}
```

There could also be proxies to this that could override where it's shown. Sometimes we want the message to show up inside a different dialog. This could intercept the message, show it locally, set an additional flag that it's already shown and push it to the parent logging hub (probably the global one) which can still write it to a file and the message log but hide it from the map canvas message bar.

Reply:
Wow: that's ambitious :)
But yes, it would be definitely useful.